client/components/social-buttons/google-one-tap.js

```diff
@@ -0,0 +1,228 @@
+import config from '@automattic/calypso-config';
+import { loadScript } from '@automattic/load-script';
+import { getLocaleSlug, localize } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+import { Component, createRef } from 'react';
+import { connect } from 'react-redux';
+import wpcomRequest from 'wpcom-proxy-request';
+import { recordTracksEventWithClientId as recordTracksEvent } from 'calypso/state/analytics/actions';
+import { isFormDisabled } from 'calypso/state/login/selectors';
+import { getErrorFromHTTPError, postLoginRequest } from 'calypso/state/login/utils';
+import { errorNotice } from 'calypso/state/notices/actions';
+import getInitialQueryArguments from 'calypso/state/selectors/get-initial-query-arguments';
+import './style.scss';
+import { OneTapLoading } from './one-tap-loading';
+
+const noop = () => {};
+
+class GoogleSocialButton extends Component {
+	buttonRef = createRef();
+	state = {
+		googleOneTapLoaded: false,
+	};
+	static propTypes = {
+		clientId: PropTypes.string.isRequired,
+		fetchBasicProfile: PropTypes.bool,
+		isFormDisabled: PropTypes.bool,
+		onClick: PropTypes.func,
+		recordTracksEvent: PropTypes.func.isRequired,
+		redirectUri: PropTypes.string,
+		responseHandler: PropTypes.func.isRequired,
+		scope: PropTypes.string,
+		startingPoint: PropTypes.string.isRequired,
+		translate: PropTypes.func.isRequired,
+		uxMode: PropTypes.string,
+	};
+
+	static defaultProps = {
+		scope: 'openid profile email',
+		fetchBasicProfile: true,
+		onClick: noop,
+	};
+	componentDidMount() {
+		if ( this.props.authCodeFromRedirect && this.props.serviceFromRedirect !== 'github' ) {
+			this.handleAuthorizationCode( {
+				auth_code: this.props.authCodeFromRedirect,
+				redirect_uri: this.props.redirectUri,
+				state: this.props.state,
+			} );
+		} else {
+			this.fetchNonceAndInitializeGoogleSignIn();
+		}
+	}
+
+	async initializeGoogleSignIn( state ) {
+		const googleSignIn = await this.loadGoogleIdentityServicesAPI();
+		if ( ! googleSignIn ) {
+			this.props.recordTracksEvent( 'calypso_social_button_failure', {
+				social_account_type: 'google',
+				starting_point: this.props.startingPoint,
+				error_code: 'google_identity_services_api_not_loaded',
+			} );
+
+			this.props.showErrorNotice(
+				this.props.translate( 'Something went wrong while trying to load Google sign-in.' )
+			);
+
+			return;
+		}
+
+		this.client = googleSignIn.initialize( {
+			client_id: this.props.clientId,
+			scope: this.props.scope,
+			ux_mode: this.props.uxMode,
+			redirect_uri: this.props.redirectUri,
+			state: state,
+			locale: getLocaleSlug(),
+			callback: ( response ) => {
+				if ( response.error ) {
+					this.props.recordTracksEvent( 'calypso_social_button_failure', {
+						social_account_type: 'google',
+						starting_point: this.props.startingPoint,
+						error_code: response.error,
+					} );
+
```

Comment:
We probably need another user-facing notice here.
In case there is an error here, the sign-in popup will be closed and nothing will happen